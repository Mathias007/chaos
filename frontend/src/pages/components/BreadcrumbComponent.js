import React, { Component } from 'react';
import { Breadcrumb } from 'antd';

const { Item } = Breadcrumb;

class BreadcrumbComponent extends Component {
    state = {}
    render() {
        return (
            <Breadcrumb style={{ margin: '16px 0' }}>
                <Item>Home</Item>
                <Item>List</Item>
                <Item>App</Item>
            </Breadcrumb>

        );
    }
}

export default BreadcrumbComponent;