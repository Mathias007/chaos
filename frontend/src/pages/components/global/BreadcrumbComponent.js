import React from "react";

import navigationTitles from "../../../_config/navigationTitles";
import styles from "../../../styles/styles";

import { Breadcrumb } from "antd";
const { Item } = Breadcrumb;

const { WORTAL, ADMINISTRATION } = navigationTitles;

export default function BreadcrumbComponent(props) {
    const { isAdminComponent, section, page } = props;
    return (
        <Breadcrumb style={styles.breadcrumb}>
            <Item>{isAdminComponent ? ADMINISTRATION : WORTAL}</Item>
            <Item>{section}</Item>
            <Item>{page}</Item>
        </Breadcrumb>
    );
}
