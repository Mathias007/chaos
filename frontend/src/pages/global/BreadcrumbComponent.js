import React, { Component } from "react";

import navigationTitles from "../../_config/navigationTitles";

import { Breadcrumb } from "antd";

const { Item } = Breadcrumb;

const { WORTAL, ADMINISTRATION } = navigationTitles;

class BreadcrumbComponent extends Component {
    state = {};
    render() {
        const { isAdminContent, section, page } = this.props;
        return (
            <Breadcrumb style={{ margin: "16px 0" }}>
                <Item>{isAdminContent ? ADMINISTRATION : WORTAL}</Item>
                <Item>{section}</Item>
                <Item>{page}</Item>
            </Breadcrumb>
        );
    }
}

export default BreadcrumbComponent;
