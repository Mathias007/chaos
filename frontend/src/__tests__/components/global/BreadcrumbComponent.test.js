import React from "react";
import renderer from "react-test-renderer";
import BreadcrumbComponent from "../../../pages/global/BreadcrumbComponent";

describe("BreadcrumbComponent test", () => {
    it("BreadcrumbComponent should match snapshot", () => {
        const component = renderer.create(<BreadcrumbComponent />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
