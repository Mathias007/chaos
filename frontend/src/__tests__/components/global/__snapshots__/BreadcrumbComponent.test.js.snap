// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbComponent test BreadcrumbComponent should match snapshot 1`] = `
<div
  className="ant-breadcrumb"
  style={
    Object {
      "margin": "16px 0",
    }
  }
>
  <span>
    <span
      className="ant-breadcrumb-link"
    >
      e-Gildia Graczy
    </span>
    <span
      className="ant-breadcrumb-separator"
    >
      /
    </span>
  </span>
</div>
`;
